### PR TITLE
Public navigation redesign

### PR DESCRIPTION
**Pull request by mikknarusk**
Screenshot 2019-04-05 at 10 54 00
Screenshot 2019-04-05 at 10 54 10
Screenshot 2019-04-05 at 10 54 21
Screenshot 2019-04-05 at 10 54 26

### CODE REVIEW

File: src/PublicNavigation/Navigation/Header/Header.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import './Header.less';
+
+const Header = ({ className, onToggle, isMenuOpen, inverse, logoLink, children }) => (
+  <div className="tw-public-navigation-menu__header pull-md-left">
+    <MenuToggle isMenuOpen={isMenuOpen} onToggle={onToggle}>
+      {children}
+    </MenuToggle>
+
+    <Logo
+      className={`pull-xs-none ${className}`}
```
Comment (KarlEST) [inline src/PublicNavigation/Navigation/Header/Header.js]: Could we move the `pull-xs-none` into the `Logo` component?

**Comment by KarlEST** [comment]
image
Goes out of the page

**Comment by domhorvathtw** [comment]
I get prop type errors when changing the language.
image

**Comment by mikknarusk** [comment]
Yeah, something is broken with changing the language in the docs page, @KarlEST is looking into it

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# v3.0.0
+## Public Navigation redesign
+
+*Breaking changes*:
+* Language selector has moved to from `Footer` to `PublicNavigation`, therefore `availableLanguages` and `onLanguageChange` now need to be passed to `PublicNavigation`, not `Footer`.
```
Comment (KarlEST) [inline CHANGELOG.md]:
```suggestion
* Language selector has moved from `Footer` to `PublicNavigation`, therefore `availableLanguages` and `onLanguageChange` now need to be passed to `PublicNavigation`, not `Footer`. 
```

File: README.md [hunk 3]
```diff
@@ -13,6 +13,23 @@ Public navigation components to be used on all [TransferWise](https://transferwi
 
 `npm install @transferwise/public-navigation`
 
+## Usage
+
+`import { PublicNavigation, Footer } from '@transferwise/public-navigation';`
+
+To get active state for nav links, pass the `activePath` prop to `PublicNavigation` so it knows which link to highlight. `activePath` should be a string which matches (`===`) an `href` inside the nav. For localised links (e.g. `/:locale/borderless`), you should include the locale inside the passed prop (so for the `de` locale you'd pass `activePath="/de/borderless"`).
+
+To get push animation behavior for mobile menu, the consuming application needs to be inside a single container with the class `navbar-push-container`. If not, the mobile menu will slide on top of the content, rather than push.
```
Comment (KarlEST) [inline README.md]: I would mention here why we prefer push one - better looking, consistent etc.

File: package.json [hunk 4]
```diff
@@ -35,7 +34,7 @@
   },
   "homepage": "https://github.com/transferwise/public-navigation#readme",
   "peerDependencies": {
-    "bootstrap": "github:transferwise/bootstrap#v5.18.2",
+    "bootstrap": "github:transferwise/bootstrap#v5.18.15",
```
Comment (KarlEST) [inline package.json]: Did you check that they moved less now?

File: src/PublicNavigation/Navigation/Header/Header.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+import './Header.less';
+
+const Header = ({ className, onToggle, isMenuOpen, inverse, logoLink, children, buttonItems }) => (
+  <div className="tw-public-navigation-menu__header pull-md-left">
+    <MenuToggle isMenuOpen={isMenuOpen} onToggle={onToggle}>
+      {children}
+    </MenuToggle>
+
+    <Logo className={`${className}`} inverse={inverse} link={logoLink} isMenuOpen={isMenuOpen} />
```
Comment (KarlEST) [inline src/PublicNavigation/Navigation/Header/Header.js]: > {`${className}`}

 🤔

Comment (KarlEST) [reply src/PublicNavigation/Navigation/Header/Header.js]:
```suggestion
    <Logo className={className} inverse={inverse} link={logoLink} isMenuOpen={isMenuOpen} />
```

File: src/PublicNavigation/Navigation/Header/Header.less [hunk 6]
```diff
@@ -0,0 +1,14 @@
+@import '~bootstrap/less/variables.less';
```
Comment (KarlEST) [inline src/PublicNavigation/Navigation/Header/Header.less]: we are not using these here

File: src/PublicNavigation/Navigation/Header/Header.spec.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+};
+
+describe('Header', () => {
+  let component;
+  beforeEach(() => {
+    component = shallow(<Header {...props} />);
+  });
+
+  it('renders menu toggle with correct props', () => {
+    const onToggle = () => {};
```
Comment (KarlEST) [inline src/PublicNavigation/Navigation/Header/Header.spec.js]: Does it matter if we use these or jest.fn()?

File: items/navigation.json [hunk 8]
```diff
@@ -1,116 +1,91 @@
 {
   "items": [
     {
-      "translationKey": "public-navigation.send",
-      "icon": "send",
+      "translationKey": "public-navigation.money-transfer",
```
[the remark this answers is not in the document]
Comment (OFFLlNE) [reply items/navigation.json]: I like this one more

File: src/PublicNavigation/Navigation/Menu/Items/Item/CardDropdown/DropdownItem/DropdownItem.spec.js [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+        titleTranslationKey="send.moneys"
+        link="https://transferwise.com/bisnes"
+        badge="new"
+      />,
+    );
+
+    expect(badge().exists()).toBe(true);
+  });
+
+  it('passes className to first DOM element', () => {
```
[the remark this answers is not in the document]
Comment (mikknarusk) [reply src/PublicNavigation/Navigation/Menu/Items/Item/CardDropdown/DropdownItem/DropdownItem.spec.js]: I know it looks weird, but it actually looks at the prop "className" passed to the DOM element, not the component. 
Had the same doubts and tried it out without passing className to the element, test breaks as intended.

File: src/PublicNavigation/Navigation/Menu/Items/Item/LanguageSelector/LanguageDropdown/LanguageDropdown.less [hunk 10]
```diff
@@ -0,0 +1,79 @@
+@import '~bootstrap/less/variables.less';
+
+.tw-public-navigation-menu__language-dropdown {
```
[the remark this answers is not in the document]
Comment (OFFLlNE) [reply src/PublicNavigation/Navigation/Menu/Items/Item/LanguageSelector/LanguageDropdown/LanguageDropdown.less]: huh, indeed. Interesting way of reusing the same class name 3 times 🤔

**Pull request merged**